interpreter/cling/lib/Interpreter/Value.cpp

```diff
@@ -255,9 +264,43 @@ namespace cling {
     const clang::ASTContext& ctx = getASTContext();
     unsigned payloadSize = ctx.getTypeSizeInChars(getType()).getQuantity();
     m_Storage.m_Ptr = AllocatedValue::CreatePayload(payloadSize, dtorFunc,
-                                                    GetNumberOfElements());
+                                                GetNumberOfElements(getType()));
+  }
+
+  void Value::AssertTypeMismatch(const char* Type) const {
```

Comment:
it is too bad that despite its name (`Assert....`) this is not a no-op when assert are disabled.

Reply:
Hm, I think its use is surrounded with `#ifndef NDEBUG` and now it is commented as it discovers other inconsistencies which we should fix at a later stage.

Reply:
We were wrong about the unused variable warning.  It will complains independently on how or if the function is used.

Reply:
Reverted the old solution.